src/Builder/Select/QueryExpressionVisitor.php

```diff
@@ -0,0 +1,199 @@
+use Solarium\Core\Query\Helper;
+use Solarium\Exception\RuntimeException;
+
+/**
+ * Query Expression Visitor.
+ *
+ * @author wicliff <wicliff.wolda@gmail.com>
+ */
+class QueryExpressionVisitor extends AbstractExpressionVisitor
+{
+    /**
+     * @var Helper
+     */
+    private $helper;
+
+    /**
+     * Constructor.
+     */
+    public function __construct()
+    {
+        $this->helper = new Helper();
+    }
+
+    /**
+     * @param \Solarium\Builder\ExpressionInterface $expression
+     *
+     * @throws \Solarium\Exception\RuntimeException
+     *
+     * @return mixed|string
+     */
+    public function walkExpression(ExpressionInterface $expression)
+    {
+        $field = $expression->getField();
+        $value = $expression->getValue()->getValue();
+
+        switch ($expression->getOperator()) {
+            case Comparison::EQ:
+            case Comparison::NEQ:
+                $strValue = $this->valueToString($value, ',', '"');
+
+                if ($value instanceof \DateTime) {
+                    $strValue = sprintf('[%1$s TO %1$s]', $strValue);
+                }
+
+                $not = (Comparison::NEQ === $expression->getOperator()) ? '-' : '';
+
+                return sprintf('%s%s:%s', $not, $field, $strValue);
+            case Comparison::GT:
+                return sprintf('%s:{%s TO *]', $field, $this->valueToString($value));
+            case Comparison::GTE:
+                return sprintf('%s:[%s TO *]', $field, $this->valueToString($value));
+            case Comparison::LT:
+                return sprintf('%s:[* TO %s}', $field, $this->valueToString($value));
+            case Comparison::LTE:
+                return sprintf('%s:[* TO %s]', $field, $this->valueToString($value));
+            case Comparison::RANGE:
+                if (\is_array($value)) {
+                    if (2 === \count($value)) {
+                        return sprintf('%s:[%s TO %s]', $field, $this->valueToString($value[0]), $this->valueToString($value[1]));
+                    }
+
+                    if (1 === \count($value)) {
+                        return sprintf('%s:[%s TO *]', $field, $this->valueToString($value[0]));
+                    }
+                }
+
+                throw new RuntimeException(sprintf('Invalid range value: %s', $value));
+            case Comparison::IN:
+                if (\is_array($value)) {
+                    return sprintf('%s:(%s)', $field, $this->valueToString($value, ' OR ', '"'));
+                }
+
+                return sprintf('%s:%s', $field, $this->valueToString($value, ',', '"'));
+            case Comparison::LIKE:
+            case Comparison::MATCH:
+                if (\is_array($value)) {
+                    return sprintf('%s:(%s)', $field, $this->valueToString($value, ' OR ', '', false));
+                }
+
+                return sprintf('%s:%s', $field, $this->valueToString($value, ',', '', false));
+            case Comparison::NIN:
```

Comment:
I do have a strong opinion about the order of the cases. I think `NIN` should come right after `IN`.

Personally, I'd stick to the same order for all the cases as the `const`s in `Comparison` are defined in.